typings/styled-components.d.ts

```diff
@@ -199,6 +199,8 @@ export interface ThemedStyledComponentsModule<T> {
 declare const styled: StyledInterface;
 
 export const css: ThemedCssFunction<any>;
+export function withTheme<C extends Component<any>>(component: C): C;
```

Comment:
I'd suggest something like this here:
```ts
export function withTheme<P extends { theme?: T; }, T>(component: Component<P>): ComponentClass<rest(P, 'theme')>;
```
where `rest(P, 'theme')` is a new type operator that is going to be landed with TS smth like 2.3, unfortunately.
I do not have any other ideas how to express removing a property from a generic type. Do you have any?